hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -26,6 +26,23 @@ const (
 type PowerVS struct {
 }
 
+func (p PowerVS) ReconcileExternalCCM(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, createOrUpdate upsert.CreateOrUpdateFN, kubeConfig *corev1.Secret, ccmImage string) (err error) {
+
+	ccm := setupCCM{ctx, c, hcluster, createOrUpdate, kubeConfig, ccmImage}
```

Comment:
Defining a type only carry all the arguments for a function call and then calling methods on it is overly OOPy and -complicated. Please either:
* Just define functions and pass their required parameter as args
* Define an `ccmOptions` type that carries the parameters and pass it as arg to a function. IMHO still overly complicated because it will need to carry the superset of all parameters needed by `setupConfigMap` and `setupDeployment`

Reply:
Moved the CCM deployment as part of CPO controller. This code is not valid now. Ptal the new code.